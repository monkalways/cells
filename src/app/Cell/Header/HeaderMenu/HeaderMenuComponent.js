import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  menuAnchorElement: PropTypes.shape({}),
  open: PropTypes.bool.isRequired,
  handleMenuOpen: PropTypes.func.isRequired,
  handleMenuClose: PropTypes.func.isRequired,
  // cellDetails: PropTypes.shape({
  //   name: PropTypes.string.isRequired,
  // }).isRequired,
  handleModalOpen: PropTypes.func.isRequired,
};

const defaultProps = {
  menuAnchorElement: null,
};

const HeaderMenu = ({
  classes,
  menuAnchorElement,
  open,
  handleMenuOpen,
  handleMenuClose,
  handleModalOpen,
}) => {
  const handleCellHistoryReportClick = () => {
    handleMenuClose();
    // history.push(`/cells/${cellDetails.name}/home/cell-history-report`);
    handleModalOpen();
  };
  return (
    <div>
      <Fab
        onClick={(event) => {
          handleMenuOpen(event);
        }}
        className={classes.icon}
        size="small"
        variant="text"
      >
        <MoreVertIcon />
      </Fab>
      <Menu
        anchorEl={menuAnchorElement}
        open={open}
        onClose={() => handleMenuClose()}
      >
        <MenuItem onClick={() => handleCellHistoryReportClick()}>
          Cell History Report
        </MenuItem>
      </Menu>
    </div>
  );
};

HeaderMenu.propTypes = propTypes;
HeaderMenu.defaultProps = defaultProps;

export default compose(
  withStyles((theme) => ({
    icon: {
      color: theme.palette.common.white,
      padding: 0,
      marginLeft: -theme.spacing.unit * 3,
    },
  })),
  withRouter,
)(HeaderMenu);
