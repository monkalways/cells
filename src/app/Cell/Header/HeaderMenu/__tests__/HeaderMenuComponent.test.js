import React from 'react';
import { shallow } from 'enzyme';
import HeaderMenuComponent from '../HeaderMenuComponent';

describe('HeaderMenuComponent', () => {
  let name;
  let handleMenuOpen;
  let handleMenuClose;
  let open;
  let menuAnchorElement;
  let push;

  beforeEach(() => {
    name = 'c1';
    handleMenuOpen = jest.fn();
    handleMenuClose = jest.fn();
    open = false;
    menuAnchorElement = {};
    push = jest.fn();
  });

  const setup = () => {
    const classes = {};
    return shallow(<HeaderMenuComponent
      classes={classes}
      menuAnchorElement={menuAnchorElement}
      open={open}
      handleMenuOpen={handleMenuOpen}
      handleMenuClose={handleMenuClose}
      history={{ push }}
      cellDetails={{ name }}
    />);
  };

  it('should render', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  // it('should handleCellCheckClick if authenticated', () => {
  //   const wrapper = setup();

  //   wrapper.find('#cellCheckButton').simulate('click');
  //   expect(push).toBeCalledWith('/cells/c1/home/cell-check');
  // });

  // it('should handleMealClick if authenticated', () => {
  //   const wrapper = setup();

  //   wrapper.find('#mealButton').simulate('click');
  //   expect(push).toBeCalledWith('/cells/c1/home/meal');
  // });

  // it('should handleMedicationClick if authenticated', () => {
  //   const wrapper = setup();

  //   wrapper.find('#medicationButton').simulate('click');
  //   expect(push).toBeCalledWith('/cells/c1/home/medication');
  // });
});
