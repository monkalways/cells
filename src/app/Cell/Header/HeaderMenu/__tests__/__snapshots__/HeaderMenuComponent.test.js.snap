// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuComponent should render 1`] = `
<withRouter(HeaderMenu)
  cellDetails={
    Object {
      "name": "c1",
    }
  }
  classes={
    Object {
      "icon": "withRouter-HeaderMenu--icon-1",
    }
  }
  handleMenuClose={[MockFunction]}
  handleMenuOpen={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  menuAnchorElement={Object {}}
  open={false}
/>
`;
